### PR TITLE
feat(Commerce Atomic): add `atomic-commerce-facet`

### PR DESCRIPTION
**Pull request by y-lakhdar**
* Add new atomic-commerce-facet component
* Following slack discussion: Went with the decision to not implement the `FacetConditionsManager`.

KIT-3171

### CODE REVIEW

**Comment by louis-bompart** [comment]
Why do we need a feature branch? 🤔

File: packages/headless/src/controllers/commerce/search-status/headless-search-status.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+export type {SearchStatusState, SearchStatus};
+
+/**
+ * Creates a `SearchStatus` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @returns A `SearchStatus` controller instance.
+ * @internal
+ * */
+export function buildSearchStatus(engine: CommerceEngine): SearchStatus {
```
Comment (olamothe) [inline packages/headless/src/controllers/commerce/search-status/headless-search-status.ts:17]: I would propose to just use `search-summary` (both for search and product listing), which expose all this state/info already. It's a repetition.

File: packages/atomic/src/components/commerce/facets/atomic-commerce-facet/atomic-commerce-facet.tsx [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+        prev.facetSearch
+      );
+    }
+    return true;
+  }
+
+  public render() {
+    return (
+      <FacetGuard
+        enabled={true}
```
[the remark this answers is not in the document]
Comment (olamothe) [reply packages/atomic/src/components/commerce/facets/atomic-commerce-facet/atomic-commerce-facet.tsx:110]: No, it's related to the facet condition manager, which headless commerce does not support.

So just hardcoding true here is the correct thing to do for now.

File: packages/atomic/src/components/commerce/facets/atomic-commerce-facet/atomic-commerce-facet.tsx [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+      <Fragment>
+        <FacetSearchInputGuard
+          canShowMoreValues={this.facetState.canShowMoreValues}
+          numberOfDisplayedValues={this.facetState.values.length}
+          withSearch={this.withSearch}
+        >
+          <FacetSearchInput
+            i18n={this.bindings.i18n}
+            label={this.displayName}
+            onChange={(value) => {
```
[the remark this answers is not in the document]
Comment (olamothe) [reply packages/atomic/src/components/commerce/facets/atomic-commerce-facet/atomic-commerce-facet.tsx:153]: No, this should ultimately be handled by CMH. 

So facet caption is not something we will need to handle client side

**Review by louis-bompart — APPROVED** [review]
lgtm gg 👍

**Review by Spuffynism — APPROVED** [review]

**Pull request merged**